avm/ptn/azd/aks/main.bicep

```diff
@@ -0,0 +1,465 @@
+param agentPools agentPoolType
+
+@description('Optional. Whether or not public network access is allowed for this resource. For security reasons it should be disabled. If not specified, it will be disabled by default if private endpoints are set and networkRuleSetIpRules are not set.  Note, requires the \'acrSku\' to be \'Premium\'.')
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+param publicNetworkAccess string = 'Enabled'
+
+@description('Optional. Scope maps setting.')
+param scopeMaps scopeMapsType
+
+@description('Optional. Specifies whether the webApplicationRoutingEnabled add-on is enabled or not.')
+param webApplicationRoutingEnabled bool?
+
+@description('Optional. Tier of your Azure container registry.')
+@allowed([
+  'Basic'
+  'Premium'
+  'Standard'
+])
+param acrSku string = 'Standard'
+
+@description('Optional. The name (as GUID) of the role assignment. If not provided, a GUID will be generated.')
+param containerRegistryRoleName string?
+
+@description('Optional. The name (as GUID) of the role assignment. If not provided, a GUID will be generated.')
+param aksClusterRoleAssignmentName string?
+
+var aksClusterAdminRole = subscriptionResourceId(
+  'Microsoft.Authorization/roleDefinitions',
+  'b1ff04bb-8a4e-4dc4-8eb5-8693973ce19b'
+)
+
+var acrPullRole = subscriptionResourceId(
+  'Microsoft.Authorization/roleDefinitions',
+  '7f951dda-4ed3-4680-a7ca-43fe172d538d'
+)
+
+var systemPoolSpec = nodePoolPresets
+
+var nodePoolPresets = {
+  vmSize: 'Standard_DS2_v2'
+  count: 3
+  minCount: 3
+  maxCount: 5
+  enableAutoScaling: true
+  availabilityZones: [
+    '1'
+    '2'
+    '3'
+  ]
+}
+
+var nodePoolBase = {
+  osType: 'Linux'
+  maxPods: 30
+  type: 'VirtualMachineScaleSets'
+  upgradeSettings: {
+    maxSurge: '33%'
+  }
+}
+
+var primaryAgentPoolProfile = [
+  union({ name: 'npsystem', mode: 'System' }, nodePoolBase, systemPoolSpec)
+]
+
+// ============== //
+// Resources      //
+// ============== //
+
+#disable-next-line no-deployments-resources
+resource avmTelemetry 'Microsoft.Resources/deployments@2024-03-01' = if (enableTelemetry) {
+  name: '46d3xbcp.ptn.azd-aks.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource logAnalytics 'Microsoft.OperationalInsights/workspaces@2021-12-01-preview' existing = if (!empty(logAnalyticsName)) {
+  name: logAnalyticsName
+}
+
+module managedCluster 'br/public:avm/res/container-service/managed-cluster:0.3.0' = {
+  name: '${uniqueString(deployment().name, location)}-managed-cluster'
+  params: {
```

Comment:
Misses passthru of `enableTelemetry` parameter

Reply:
Not the correct fix. Please don't add the telemetry parameter as a condition, but just pass the parameter in the params block through to the linked module. Also please don't self-resolve comments 😉

Reply:
I have updated by your comments.